numba/numpy_support.py

```diff
@@ -123,6 +123,8 @@ def as_dtype(nbtype):
     if isinstance(nbtype, (types.CharSeq, types.UnicodeCharSeq)):
         letter = _as_dtype_letters[type(nbtype)]
         return numpy.dtype('%s%d' % (letter, nbtype.count))
+    if isinstance(nbtype, types.Record):
```

Comment:
This is not a required change.  Just that in some of my exploration, I wanted to call `as_dtype(my_numba_record)` for creating a a structure dtype.  Also, this will make `my_numba_record==from_dtype(as_dtype(my_numba_record))` work.